packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -333,62 +381,43 @@ const SideNav = ({
       sideNavItems.map((item) => ({
         ...item,
         children: item.children?.map((childProps) => (
-          <SideNavItemContent
+          <SideNavItemContentContext.Provider
+            value={{ isCompact: isCompact }}
             key={childProps.id}
-            scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
-            {...childProps}
-          />
+          >
+            <SideNavItemContent
+              key={childProps.id}
+              scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
+              {...childProps}
+            />
+          </SideNavItemContentContext.Provider>
         )),
       })),
-    [getRefIfThisIsFirstNodeWithIsSelected, sideNavItems],
+    [getRefIfThisIsFirstNodeWithIsSelected, sideNavItems, isCompact],
   );
 
-  const sideNavCollapseHandler = useCallback(() => {
-    setSideNavCollapsed(!isSideNavCollapsed);
-    onCollapse?.();
-  }, [isSideNavCollapsed, setSideNavCollapsed, onCollapse]);
-
   const sideNavExpandClickHandler = useCallback(() => {
+    isSideNavCollapsed ? onExpand?.() : onCollapse?.();
```

Comment:
I wonder if we should set default values for these functions so we don't have to do `?.` everywhere.

But you can't just do use a default value in the arg:
```ts
// WRONG
export const MyComponent = ({ onExpand = () => {} }) => <div />
```

But you can't just do use a default value in the arg:
```ts
export const defaultProps: Pick<Required<MyComponentProps>, "onExpand"> = {
  onExpand: () => {},
}

const MyComponent = ({ onExpand = defaultProps.onExpand }) => <div />
```

This way, it doesn't re-create that function each time a render occurs; forcing a re-render in the DOM.